python/tank_vendor/shotgun_api3/shotgun.py

```diff
@@ -4090,7 +4090,13 @@ def _upload_data_to_storage(self, data, content_type, size, storage_url):
                     if e.code == 503:
                         raise ShotgunError("Got a 503 response when uploading to %s: %s" % (storage_url, e))
                     raise ShotgunError("Unanticipated error occurred uploading to %s: %s" % (storage_url, e))
-
+            except urllib.error.URLError as e:
```

Comment:
Could you try to restrict the error to a specific case/code/errno?

I think there are multiple URLError that would be persistent and so don't need to be re-attempted

Reply:
I just added the errno filter, and added a log of the storage_url

Reply:
Looks good. Is there a special ERRNO constant we can use instead of hardcoding 10060? 
Sometimes the Python packages provide such constants.